### ISSUE TITLE
I cannot understand why it is different?

### ISSUE THREAD

**Issue by pismute** [chaijs/chai#344]
I am developing some test cases using array.slice() method, but I don't understand my results.

envs:

- node: v0.10.33
- mocha: v2.0.1
- chai: v1.10.0

It is ok:

```js
var a = [1,2];                                                                                                                                  
expect(a.slice()).to.deep.equal(a);                                                                                                             
expect(a.slice()).to.not.equal(a);  
```

but It is not:

```js
var a = [1,2];                                                                                                                                  
expect(a.slice()).to.deep.equal(a).and.not.equal(a);
```
I have got an error like that:

```
AssertionError: expected [ 1, 2 ] to not deeply equal [ 1, 2 ]
```

**Comment by keithamus** [comment]
Hey @pismute thanks for the issue. I think I know whats going on - it all has to do with how Chai's flags work.

Chai's properties set flags on the assertion - `.deep` sets `deep = true`, `.not` sets `negate = true`. These flags help the final assertion (such as `.equal`) decide how what to do. So when you do `expect(a.slice()).deep` you're setting the `deep` flag to `true`.

Now, `.and` copies over all flags from the previous assertion and applies it to the new one. It does this so that it can make the same kind of assertions in a chain - the idea being that you set up a config for one assertion, and use chain to assert with the flags from previous ones. I can show you how this works by opening up a node terminal and doing this:

```js
> chai.expect([1,2]).deep
{ 
  "__flags": {
    "object": [
      1,
      2
    ],
    "deep": true
  }
}

> chai.expect([1,2]).deep.and
{
  "__flags": {
    "object": [
      1,
      2
    ],
    "deep": true
  }
}

```

So your assertion `expect(a.slice()).to.deep.equal(a).and.not.equal(a);` is actually saying `expect(a.slice()).to.deep.equal(a).and.not.deep.equal(a);`. Obviously `[1,2]` deeply equals `[1,2]` so this fails, with the error that you saw.

If you indent your syntax a little, you can see it makes things a bit clearer:

```js
expect([1,2].slice())
  .to.deep
    .equal([1,2])
   .and.not
      .equal([1,2])
```

In summary, flags like `.not` and `.deep` carry over with `.and`. This means the best way to write your assertions is as two assertions:

```js
var a = [1,2];                                                                                                                                  
expect(a.slice()).to.deep.equal(a);                                                                                                             
expect(a.slice()).to.not.equal(a);
```

**Comment by pismute** [comment]
Thank you very much.
